### PR TITLE
Consensus

### PR DESCRIPTION
**Pull request by zjsunzone**

### CODE REVIEW

File: consensus/cbft/peer.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	"github.com/PlatONnetwork/PlatON-Go/p2p"
+	"github.com/PlatONnetwork/PlatON-Go/p2p/discover"
+	mapset "github.com/deckarep/golang-set"
+)
+
+var (
+	errClosed                  = errors.New("peer set is closed")
+	errAlreadyRegistered       = errors.New("peer is already registered")
+	errNotRegistered           = errors.New("peer is not registered")           // errNotRegistered represents that the node is not registered.
+	errInvalidHandshakeMessage = errors.New("Invalid handshake message params") // The parameters passed in the node handshake are not correct.
```
Comment (TraceBundy) [inline consensus/cbft/peer.go]: 小写

File: consensus/cbft/router/peer.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+	ProtocolVersion int `json:"protocol_version"`
+	HighestQCBn     int `json:"highest_qc_bn"`
+	LockedBn        int `json:"locked_bn"`
+	CommitBn        int `json:"commit_bn"`
+}
+
+// Info output status information of the current peer.
+func (p *peer) Info() *PeerInfo {
+	pv, qc, locked, commit := p.version, p.highestQCBn.Int64(), p.lockedBn.Int64(), p.commitBn.Int64()
+	return &PeerInfo{
```
Comment (mowind) [inline consensus/cbft/router/peer.go]: suggest use `uint64` instead of `int`

File: consensus/cbft/router/peer.go [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+// Peer represents a node in the network.
+type peer struct {
+	*p2p.Peer
+	id      string
+	rw      p2p.MsgReadWriter
+	version int           // Protocol version negotiated
+	term    chan struct{} // Termination channel to stop the broadcaster
+
+	// Node status information
+	highestQCBn *big.Int
```
Comment (mowind) [inline consensus/cbft/router/peer.go]: use one lock is enough

File: consensus/cbft/router/router.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+const DEFAULT_FAN_OUT = 5
+
+// Router implements the message protocol of gossip.
+//
+// 1.Responsible for picking receiving nodes based on message type.
+// 2.Generate a random node list based on fan-out.
+// 3.Duplicate verification of messages.
+type router struct {
+	bft    Cbft                          // Implementation of cbft interface
+	filter func(*peer, common.Hash) bool // Used for filtering node
```
Comment (mowind) [inline consensus/cbft/router/router.go]: `func(*peer, common.Hash) bool` use multiple times, should define as a type.

**Pull request merged**